### PR TITLE
fix(setup): do not create field which already exist

### PR DESCRIPTION
**Pull request by sumo-drosiek**
If we get field which is non lowercase, we try to recreate it, which leads to errors and setup process fail

Fixes https://github.com/SumoLogic/sumologic-kubernetes-collection/issues/2865 proper way

### Checklist

- [ ] Changelog updated or skip changelog label added
- [ ] Documentation updated
- [ ] Template tests added for new features
- [ ] Integration tests added or modified for major features

### CODE REVIEW

**Comment by swiatekm-sumo** [comment]
What does this change actually do? It seems like it imports the field even if it's uppercase - but then won't we actually lowercase it when applying? That doesn't sound like the right thing to do here.

**Comment by sumo-drosiek** [comment]
> What does this change actually do? It seems like it imports the field even if it's uppercase - but then won't we actually lowercase it when applying? That doesn't sound like the right thing to do here.

You may be right. I will try to research the following solution:
change `var.create_fields` to be a map, so every field can be managed separately. WDYT?

**Comment by swiatekm-sumo** [comment]
> > What does this change actually do? It seems like it imports the field even if it's uppercase - but then won't we actually lowercase it when applying? That doesn't sound like the right thing to do here.
> 
> You may be right. I will try to research the following solution: change `var.create_fields` to be a map, so every field can be managed separately. WDYT?

I think that if we're going to require special configuration for this to work, the user may as well just make the field uppercase in their values.yaml. Unless I'm misunderstanding your suggestion.

> I found that we can simply ignore name changes: https://developer.hashicorp.com/terraform/language/meta-arguments/lifecycle#ignore_changes

That would actually work, I think. I'm a bit hesitant about this change in general, because I don't think we have a way of easily testing it. I don't think you can create an uppercase field in Sumo right now.

**Comment by sumo-drosiek** [comment]
> That would actually work, I think. I'm a bit hesitant about this change in general, because I don't think we have a way of easily testing it. I don't think you can create an uppercase field in Sumo right now.

I was thinking about it and I will manually import some field with different name and see what will happen during plan/apply

**Comment by sumo-drosiek** [comment]
Tested:

```
➜  terraform terraform apply # with ignore_changes
sumologic_field.cluster: Refreshing state... [id=zzz]
sumologic_field.cluster2: Refreshing state... [id=aaa]

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no
differences, so no changes are needed.

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
➜  terraform terraform plan # without ignore changes
sumologic_field.cluster: Refreshing state... [id=zzz]
sumologic_field.cluster2: Refreshing state... [id=aaa]

Terraform used the selected providers to generate the following execution plan. Resource
actions are indicated with the following symbols:
-/+ destroy and then create replacement

Terraform will perform the following actions:

  # sumologic_field.cluster2 must be replaced
-/+ resource "sumologic_field" "cluster2" {
      ~ field_id   = "zzz" -> (known after apply)
      ~ field_name = "service" -> "clusters" # forces replacement
      ~ id         = "aaa" -> (known after apply)
        # (2 unchanged attributes hidden)
    }

Plan: 1 to add, 0 to change, 1 to destroy.
```

File: deploy/helm/sumologic/conf/setup/fields.tf [hunk 1]
```diff
@@ -2,6 +2,10 @@
 resource "sumologic_field" {{ $value | quote }} {
   count = var.create_fields ? 1 : 0
 
+  lifecycle {
+    ignore_changes = [field_name, data_type]
+  }
```
[the remark this answers is not in the document]
Comment (sumo-drosiek) [reply deploy/helm/sumologic/conf/setup/fields.tf]: Added comment with description and link to https://github.com/SumoLogic/sumologic-kubernetes-collection/issues/2865

**Review by swiatekm-sumo — APPROVED** [review]

**Pull request merged**